src/database/data.js

```diff
@@ -41,34 +40,44 @@ export default class DataConnection extends React.Component {
   // implemented `componentShouldUpdate`
```

Comment:
How would that work? Like what do we put in `componentShouldUpdate`  in this case, so that i renders when we want to?

Reply:
well, here's the weird part, we basically never want this component to update, the only time would be if we were feeding in new props via code, also, i got this wrong, it's `shouldComponentUpdate` 😞 , but the idea is if for some reason we receive a new `location` prop, we would tear down all the old handlers and set up new ones. i should remove this comment now that it's no longer necessary